react-backend/client/src/components/admin/adminUserDetailsPage.js

```diff
@@ -0,0 +1,396 @@
+          </div>
+          <h2>Edit {this.state.userFullName}s Details</h2>
+          {this.state.errorMessage && (
+            <span className="ErrorText">{this.state.errorMessage}</span>
+          )}
+
+          <form>
+            <label className="Form__Label">
+              Edit Full Name
+              <input
+                className="Form__Input"
+                type="text"
+                name="userFullName"
+                onChange={this.handleChange}
+                value={this.state.userFullName}
+              />
+            </label>
+            <label className="Form__Label">
+              Edit Sex
+              <select className="Form__Input" onChange={this.handleChangeSex}>
+                <option defaultValue value={this.state.sex}>
+                  Change sex: {this.state.sex}
+                </option>
+                <option value="prefer not to say">prefer not to say</option>
+                <option value="male">male</option>
+                <option value="female">female</option>
+              </select>
+            </label>
+            <label className="Form__Label">
+              Edit Year of Birth
+              <input
+                type="text"
+                className="Form__Input"
+                name="yearOfBirth"
+                onChange={this.handleChange}
+                value={this.state.yearOfBirth}
+              />
+            </label>
+
+            <label className="Form__Label">
+              Edit Email
+              <input
+                className="Form__Input"
+                type="text"
+                name="email"
+                onChange={this.handleChange}
+                value={this.state.email}
+              />
+            </label>
+            {this.state.successMessage ===
+              'The user details have been successfully updated' && (
+              <span className="SuccessText">{this.state.successMessage}</span>
+            )}
+            <button className="Button" onClick={submitHandler}>
+              Submit
+            </button>
+          </form>
+
+          <Link to="/admin/users">
+            <button className="Button ButtonBack">Back to all users</button>
+          </Link>
+          <br />
+          <Logoutbutton
+            updateLoggedIn={this.props.updateLoggedIn}
+            redirectUser={this.props.history.push}
+          />
+          <br />
+        </div>
+        <div className="visible-printer qr-code-to-print">
+          <div className="dashed">
+            <img height="182" src={qrcodelogo} alt="Power to change Logo" />
+            <img
+              className="QR__image"
+              src={this.state.url}
+              alt="This is your QRcode"
+            />
+            <h5>
+              Please print this QR code and <br /> bring it with you to access
+              next time
+            </h5>
+          </div>
+        </div>
+      </div>
+    );
+  };
+
+  renderNotAuthenticated = () => {
```

Comment:
This could for sure be a reusable component. Maybe let's just leave it until the great refactor.

Reply:
I'm really looking forward to this massive refactor :)